lib/collection/src/hash_ring.rs

```diff
@@ -199,33 +201,36 @@ impl<T: Eq + Hash> HashRingRouter<T> {
                 );
 
-                match from.cmp(&to) {
-                    Ordering::Equal => 1.0,
-                    // Resharding up:
-                    //
-                    // - shards: 1 -> 2
-                    //   points: 100 -> 50/50
-                    //   transfer fraction of each shard: 1/2/1 = 0.5
-                    // - shards: 2 -> 3
-                    //   points: 50/50 -> 33/33/33
-                    //   transfer fraction of each shard: 1/3/2 = 0.167
-                    // - shards: 3 -> 4
-                    //   points: 33/33/33 -> 25/25/25/25
-                    //   transfer fraction of each shard: 1/4/3 = 0.083
-                    Ordering::Less => (1.0 / to as f32) / (from as f32),
-                    // Resharding down:
-                    //
-                    // - shards: 2 -> 1
-                    //   points: 50/50 -> 100
-                    //   transfer fraction of each shard: 1/1 = 1.0
-                    // - shards: 3 -> 2
-                    //   points: 33/33/33 -> 50/50
-                    //   transfer fraction of each shard: 1/2 = 0.5
-                    // - shards: 4 -> 3
-                    //   points: 25/25/25/25 -> 33/33/33
-                    //   transfer fraction of each shard: 1/3 = 0.333
-                    Ordering::Greater => 1.0 / to as f32,
-                }
+                // Resharding up:
+                //
+                // - shards: 1 -> 2
+                //   points: 100 -> 50/50
+                //   transfer points of each shard: 50/1 = 50 -> 50/100 = 50%
+                //   transfer fraction of each shard: 1/to = 1/2 = 0.5
+                // - shards: 2 -> 3
+                //   points: 50/50 -> 33/33/33
+                //   transfer points of each shard: 33/2 = 16.5 -> 16.5/50 = 33%
+                //   transfer fraction of each shard: 1/to = 1/3 = 0.33
+                // - shards: 3 -> 4
+                //   points: 33/33/33 -> 25/25/25/25
+                //   transfer points of each shard: 25/3 = 8.3 -> 8.3/33 = 25%
+                //   transfer fraction of each shard: 1/to = 1/4 = 0.25
+                //
+                // Resharding down:
+                //
+                // - shards: 2 -> 1
+                //   points: 50/50 -> 100
+                //   transfer points of each shard: 50/1 = 50 -> 50/50 = 100%
+                //   transfer fraction of each shard: 1/to = 1/1 = 1.0
+                // - shards: 3 -> 2
+                //   points: 33/33/33 -> 50/50
+                //   transfer points of each shard: 33/2 = 16.5 -> 16.5/33 = 50%
+                //   transfer fraction of each shard: 1/to = 1/2 = 0.5
+                // - shards: 4 -> 3
+                //   points: 25/25/25/25 -> 33/33/33
+                //   transfer points of each shard: 25/3 = 8.3 -> 8.3/25 = 33%
+                //   transfer fraction of each shard: 1/to = 1/3 = 0.33
+                1.0 / to as f32
```

Comment:
```suggestion
                // - shards: 2 -> 1
                //   points: 50/50 -> 100
                //   transfer points of each shard: 50/1 = 50 -> 50/50 = 100%
                //   transfer fraction to each shard: 1/to = 1/1 = 1.0
                // - shards: 3 -> 2
                //   points: 33/33/33 -> 50/50
                //   transfer points of each shard: 33/2 = 16.5 -> 16.5/33 = 50%
                //   transfer fraction to each shard: 1/to = 1/2 = 0.5
                // - shards: 4 -> 3
                //   points: 25/25/25/25 -> 33/33/33
                //   transfer points of each shard: 25/3 = 8.3 -> 8.3/25 = 33%
                //   transfer fraction to each shard: 1/to = 1/3 = 0.33
                1.0 / to as f32
```

nitpick comment. Rest LGTM.

Reply:
Also `Ordering::Equal => 1.0` isn't accounted for. But I guess that's fine because we don't expected resharding `old.len()` and `new.len()` to ever be same here.

Reply:
Yeah. You know what, I'll add that case explicitly anyway to be on the safe side.